### PR TITLE
8004 drop down update

### PR DESCRIPTION
**Pull request by kialj876**
*Issue #:* /bcgov/entity#8004

*Description of changes:*
see ticket for details

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/interfaces/store-interfaces/state-interfaces/account-information-interface.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Account information interface
+import { AccountProductSubscriptionIF } from '@/interfaces'
```
Comment (jordiwes) [inline ppr-ui/src/interfaces/store-interfaces/state-interfaces/account-information-interface.ts]: why this import?

File: ppr-ui/tests/unit/RegistrationBarTypeAhead.spec.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    wrapper = createComponent()
+  })
+  afterEach(() => {
+    wrapper.destroy()
+  })
+
+  it('renders registration autocomplete list', async () => {
+    expect(wrapper.findComponent(RegistrationBarTypeAheadList).exists()).toBe(true)
+    expect(wrapper.find(registrationTypeAhead).exists()).toBe(true)
+    // check autocomplete displayed
```
Comment (jordiwes) [inline ppr-ui/tests/unit/RegistrationBarTypeAhead.spec.ts:85]: should you check that autocomplete is displayed here :)

File: ppr-ui/tests/unit/test-data/mock-registration-responses.ts [hunk 3]
```diff
@@ -29,22 +27,22 @@ export const mockedRegistration2: RegistrationSummaryIF = {
...
-    documentId: 'D9000338',
-    baseRegistrationNumber: '',
-    registrationType: APIRegistrationTypes.MISCELLANEOUS_OTHER,
-    registrationDescription: '',
-    path: '/path/to/doc',
-    createDateTime: '2021-08-01T17:21:17+00:00',
-    clientReferenceId: ''
+  documentId: 'D9000338',
+  baseRegistrationNumber: '',
+  registrationType: APIRegistrationTypes.OTHER,
```
Comment (jordiwes) [inline ppr-ui/tests/unit/test-data/mock-registration-responses.ts:43]: Should we add a description here? I guess this begs a bigger picture question... if the user fills out the dialog with their own registration description, how does this get sent to the API?

File: ppr-ui/tests/unit/RegistrationBarTypeAhead.spec.ts (hunk 2 above)
Comment (kialj876) [reply ppr-ui/tests/unit/RegistrationBarTypeAhead.spec.ts:85]: Oh I will move that comment above (the *registrationTypeAhead* is the autocomplete). I still couldn't access the actual list via jest though :(

File: ppr-ui/tests/unit/test-data/mock-registration-responses.ts (hunk 3 above)
Comment (kialj876) [reply ppr-ui/tests/unit/test-data/mock-registration-responses.ts:43]: so when they enter the name in the modal it is saved to the store in the new *registrationTypeOtherDesc* . Doug said he would add *otherDescription* as a field in the call that the api is expecting so we can add that in to the post (I didn't touch that since this code is supposed to just be for the dropdown)

Comment (kialj876) [reply ppr-ui/tests/unit/test-data/mock-registration-responses.ts:43]: I didn't add it to the registrationType object because it is a bit of an edge case since it only applies to 1 out of the 20 or so types

File: ppr-ui/src/interfaces/store-interfaces/state-interfaces/account-information-interface.ts (hunk 1 above)
Comment (kialj876) [reply ppr-ui/src/interfaces/store-interfaces/state-interfaces/account-information-interface.ts]: oops, I was going to put this in here and then decided to have it as it's own section in the store. Will remove

**Comment by jordiwes** [comment]
resolve the conflicts then good to go

**Review by jordiwes — APPROVED** [review]
